packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1292,6 +1136,194 @@ export class TransactionController extends BaseController<
     return latestIncomingTxBlockNumber;
   }
 
+  private async processApproval(
+    transactionMeta: TransactionMeta,
+  ): Promise<string> {
+    const transactionId = transactionMeta.id;
+    let rejected = false;
+    let resultCallbacks: AcceptResultCallbacks | undefined;
+
+    try {
+      const acceptResult = await this.requestApproval(transactionMeta);
+      resultCallbacks = acceptResult.resultCallbacks;
+
+      const updatedMeta = this.getTransaction(transactionId);
+
+      const isCompleted =
+        updatedMeta && this.isLocalFinalState(updatedMeta.status);
+
+      if (updatedMeta && !isCompleted) {
+        await this.approveTransaction(transactionId);
+      }
+    } catch (error: any) {
+      const updatedMeta = this.getTransaction(transactionId);
+
+      const isCompleted =
+        updatedMeta && this.isLocalFinalState(updatedMeta.status);
+
+      if (updatedMeta && !isCompleted) {
+        if (error.code === errorCodes.provider.userRejectedRequest) {
+          this.cancelTransaction(transactionId);
+          rejected = true;
+        } else {
+          this.failTransaction(updatedMeta, error);
+        }
+      }
+    }
+
+    if (rejected) {
+      throw ethErrors.provider.userRejectedRequest(
+        'User rejected the transaction',
+      );
+    }
```

Comment:
rejected only seems to be used once.. Can we remove that var and just throw within the previous if clause?

```js
      if (updatedMeta && !isCompleted) {
        if (error.code === errorCodes.provider.userRejectedRequest) {
          this.cancelTransaction(transactionId);
          throw ethErrors.provider.userRejectedRequest(
            'User rejected the transaction',
          );
        } else {
          this.failTransaction(updatedMeta, error);
        }
      }
    }
```

Reply:
Done.